packages/hashi/src/mediator.js

```diff
@@ -50,6 +52,38 @@ class Mediator {
     this.remote.postMessage(message, '*');
   }
 
+  // a function to manage messages for kolibri.js,
+  // when most messages require a response, to minimize redundancy
+  sendMessageAwaitReply({ event, data, nameSpace }) {
+    return new Promise((resolve, reject) => {
+      const msgId = uuidv4();
+      function handler(message) {
+        if (message.message_id === msgId && message.type === 'response') {
+          if (message.status === 'success') {
+            console.log(message.data);
+            return resolve(message.data);
+          } else if (message.status === 'failure' && message.err) {
+            return reject(message.err);
+          }
+          // Otherwise something unspecified happened
+          return reject();
+        }
+        this.removeMessageHandler({
```

Comment:
This is only removing the message handler in the case when the message *doesn't* match - this might cause weird behaviour. Should move this inside the `if` block as the first thing.